### ISSUE TITLE
MissingFieldException when using abstract base class with default values

### ISSUE THREAD

**Issue by thumannw** [Kotlin/kotlinx.serialization#1563]
Steps to reproduce:
1. Checkout [this](https://github.com/thumannw/kotlinx.serialization-MissingFieldException) repo and import in IntelliJ.
2. If not a clean build, run the Gradle clean task. 
3. Run `Test.main` in IntelliJ. Should run through successfully.
4. Now uncomment line 6 in `Other.kt`.
5. Do _not_ perform a clean build, just rerun `Test.main` in IntelliJ. The exception below should be thrown.
6. After a Gradle clean, the main method should run through successfully again.

```
Exception in thread "main" kotlinx.serialization.MissingFieldException: Fields [first, second] are required for type with serial name 'Derived', but they were missing
	at kotlinx.serialization.internal.PluginExceptionsKt.throwMissingFieldException(PluginExceptions.kt:20)
	at Derived.<init>(Test.kt:10)
	at Derived$$serializer.deserialize(Test.kt:10)
	at Derived$$serializer.deserialize(Test.kt:10)
	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:63)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:32)
	at kotlinx.serialization.json.Json.decodeFromString(Json.kt:95)
	at TestKt.main(Test.kt:6)
	at TestKt.main(Test.kt)
```

**Comment by thumannw** [comment]
Seems to be an issue with incremental building (note that the base class is located in a different file). However, in a real life project I managed to provoke the MissingFieldException also on a clean build (I was not able to extract this into a minimal reproducible example though). So maybe the compilation order plays a role here in some way?

**Comment by thumannw** [comment]
Related to #1481

**Comment by qwwdfsad** [comment]
Closing as fixed in 1.6.0
